### ISSUE TITLE
Scope only for given component

### ISSUE THREAD

**Issue by MichalJakubeczy** [kuuurt13/ng-block-ui#123]
I have multiple components of the same type on a page where following *blockUI attribute is defined.
`<table *blockUI="'results-table'">`

In the ts file I reference BlockUI like this:
`@BlockUI('results-table') public blocker: NgBlockUI;`

The problem is that when I trigger
`blocker.start('Please wait')`

All of components become 'busy' (because of the same identifier). But I need to make busy only the one in a particular instance. Yes, they can be distinguished by various names, but it is the same component.

Is there any possibility to trigger start only for a blocker on a current instance?

**Comment by kuuurt13** [comment]
@MichalJakubeczy currently there isn't anything built into the library to only target a current instance but if I am following you correctly this can be achieved with a small amount of code. See the below example and let me know if this works for what you are trying to do.

https://stackblitz.com/edit/ng-block-ui-unique-block-name?embed=1&file=src/app/custom-table/custom-table.component.ts

**Comment by MichalJakubeczy** [comment]
@kuuurt13 thanks for an example. Yes, this helps, thanks, you can close the issue/question.

Anyway, do you plan to integrate targetting only a current instance?

**Comment by kuuurt13** [comment]
@MichalJakubeczy, this seems like a semi common use case for the library so I think this is something we could add. 

I'm thinking maybe we could add a second parameter to the decorator and we could have a setting that when set to `true` would "scope" it to that instance. Then in the template, you would pass the name to the  `blockUI` decorator so the correct instance is targeted. I think this should be possible but I would have to do a little research first. In the meantime, let me know what you think.

#### Component
```ts
@Component({
  selector: 'custom-table',
  templateUrl: 'custom-table.component.html'
})
export class CustomTableComponent {
  // Flag the block ui instance as scoped 
  @BlockUI('table', { isScoped: true }) blockUI: NgBlockUI;
}
```

#### Template
```html
<!-- Pass the blockUI name so the correct instance is targeted -->
<table *blockUI="blockUI.name">
  <tr>
    <th>First Name</th>
    <th>Last Name</th>
  </tr>
  <tr>
    <td>John</td>
    <td>Doe</td>
  </tr>
</table>
```

**Comment by MichalJakubeczy** [comment]
@kuuurt13 yes, this would definitely help. What does the `table` exactly mean in the first parameter of a decorator? Is it element name? 

Also it would be great to be still able to define a name for the component like this:

**Template**
```
<!-- Pass the blockUI name so the correct instance is targeted -->
<table *blockUI="'results-table'">
  <tr>
    <th>First Name</th>
    <th>Last Name</th>
  </tr>
  <tr>
    <td>John</td>
    <td>Doe</td>
  </tr>
</table>
```

**Component**
```
@Component({
  selector: 'custom-table',
  templateUrl: 'custom-table.component.html'
})
export class CustomTableComponent {
  // Flag the block ui instance as scoped 
  @BlockUI('results-table', { isScoped: true }) blockUI: NgBlockUI;
}
```

But if it means more work also your proposed solution is fine.

**Comment by kuuurt13** [comment]
@MichalJakubeczy `table` (or in your example `results-table`) is the name for blockUI to target when dispatching the start/stop events. So when you call `this.blockUI.start()` it will dispatch a start event targeting `table`. Any blockUI directives that are passed `table` as its name (in any template) will then start blocking (`*blockUI="'table'"`). The reason you ran into your issue is that all your components were using the same name and currently blockUI uses the same instance for all components so all were started when you called start. 

So what needs to happen is a unique name needs to be created for each instance of a component and then that unique name needs to be passed to the directive. That way when start is called it will use the unique name that both the component and its template share. So what I want to do is that when you pass `{ isScoped: true }` the library will create a unique block name with `table` such as `table-<unique-guid>` and you will have to pass `blockUI.name` to the directive so the blockUI instance and the directive share the same unique name.

Anyway, I have started working on this and I have an initial proof of concept working so I should be able to roll this into the upcoming `3.0.0` release, I will keep you posted.

**Comment by MichalJakubeczy** [comment]
@kuuurt13 yes, that's great. Thanks a lot. When do you plan to release 3.0.0 (will it be Angular 9 compatible)?

**Comment by kuuurt13** [comment]
Fixed by #113
